dCommon/Amf3.h

```diff
@@ -106,7 +94,7 @@ using AMFDoubleValue = AMFValue<double>;
  */
 class AMFArrayValue : public AMFBaseValue {
 	using AMFAssociative =
-		std::unordered_map<std::string, std::unique_ptr<AMFBaseValue>, GeneralUtils::transparent_string_hash, std::equal_to<>>;
+		std::unordered_map<std::string, std::unique_ptr<AMFBaseValue>, GeneralUtils::transparent_string_hash, std::equal_to<void>>;
```

Comment:
lets get this in sooner rather than later in a separate PR because its breaking on gcc10, a supported compiler

Reply:
Unfortunately, this still isn't supported with gcc 10. I know I said earlier in the Discord, but I was mistaken. We either need to up the compiler requirement or strip out the offending code.

Reply:
I believe there are two ways we can go forward with this fix:
option A is to remove the transparent string hashing and use references to strings for comparisons (I dont think there is much copying if any going on when looking elements up and these strings are generally not big enough to cause large slowdowns?).
option B is to try and use some older stuff and make it compliant with gcc.

Why did you implement the string hasher initially? Does it make that much of a performance impact we'll notice at runtime?

Reply:
I doubt it'll make a noticeable difference, but it was good practice and I thought it might come in useful for other unordered_maps.

One way we could tackle this is by gating it behind an #ifdef
```suggestion
#if defined(__cpp_lib_generic_unordered_lookup) && __cpp_lib_generic_unordered_lookup >= 201811L
		std::unordered_map<std::string, std::unique_ptr<AMFBaseValue>, GeneralUtils::transparent_string_hash, std::equal_to<void>>;
#else
		std::unordered_map<std::string, std::unique_ptr<AMFBaseValue>, std::equal_to<void>>;
#endif		
```

Reply:
I'm not particularly a fan of changing a container definition based on the os you are on, however a solution is a solution.  If you're happy with this feel free to make a pr and we'll merge it :).  I do believe your set for the equal_to template is one argument early (the standard docs show it as argument 4 and not argument 3) so i would make sure this works.

Reply:
I was doubly wrong. :(
The issue at hand is that std::unordered_map::find() was changed to use a template argument with C++20. For whatever reason, this particular feature of C++20 was not implemented in gcc 10. Thus, we can either rollback the change, or we can up the version requirement to gcc 11. Unfortunately, therre doesn't seem to be any other option.

Reply:
I still leave it to your discretion of what to do.  which ever method you choose i'll support it.

Reply:
Undoing the changes would require either changing all the string_views to strings or creating temporary strings at the call sites to handle the hashing. I think it makes more sense to just say we require versions of gcc > 10

Reply:
gcc 11 is fine on my end.  again, im not concerned about copies for amf as these strings are generally small enough for small string optimization and speed is a far greater factor in other spots of the codebase.

Reply:
It's not the creation of the strings that bothers me as much as the fact we'd need to put constructors in every call of find(). 😅

Reply:
Doesn't seem to. If you try and compile this code on gcc 10, find complains about the argument being the wrong type. It doesn't seem to do any implicit conversion.

Reply:
odd